contracts/libs/0.8.x/BytecodeStorageV1.sol

```diff
@@ -31,13 +38,37 @@ library BytecodeStorage {
     //---------------------------------------------------------------------------------------------------------------//
     // 0              | N/A  | 0            |                                                                        //
     // 0              | 1    | 1            | single byte opcode for making the storage contract non-executable      //
-    // 1              | 32   | 33           | the 32 bytes for storing the deploying contract's (0-padded) address   //
+    // 1              | 32   | 33           | the 32 byte slot used for storing a basic versioning string            //
+    // 33             | 32   | 65           | the 32 bytes for storing the deploying contract's (0-padded) address   //
     //---------------------------------------------------------------------------------------------------------------//
-    // Define the offset for where the "logic bytes" end, and the "data bytes" begin. Note that this is a manually
+    // Define the offset for where the "meta bytes" end, and the "data bytes" begin. Note that this is a manually
     // calculated value, and must be updated if the above table is changed. It is expected that tests will fail
     // loudly if these values are not updated in-step with eachother.
-    uint256 internal constant DATA_OFFSET = 33;
-    uint256 internal constant ADDRESS_OFFSET = 1;
+    uint256 internal constant VERSION_OFFSET = 1;
+    uint256 internal constant ADDRESS_OFFSET = 33;
+    uint256 internal constant DATA_OFFSET = 65;
+
+    // Define the set of known *historic* offset values for where the "meta bytes" end, and the "data bytes" begin.
+    uint256 internal constant V0_ADDRESS_OFFSET = 72;
+    uint256 internal constant V0_DATA_OFFSET = 104;
+    uint256 internal constant V1_ADDRESS_OFFSET = ADDRESS_OFFSET;
+    uint256 internal constant V1_DATA_OFFSET = DATA_OFFSET;
+
+    // Define the set of known valid version strings that may be stored in the deployed storage contract bytecode
+    // note: These are all intentionally exactly 32-bytes and are null-terminated
+    // Used for storage contracts that were deployed by an unknown source
+    bytes32 internal constant UNKNOWN_VERSION_STRING =
+        "UNKNOWN_VERSION_STRING_________ ";
```

Comment:
nit: can you add a dev comment to give a bit more context as to why these need to be null-terminated?

Reply:
Sure thing! Done.